packages/web/src/components/basic/ReactiveComponent.js

```diff
@@ -73,6 +74,27 @@ class ReactiveComponent extends Component {
 		) {
 			nextProps.onAllData(parseHits(nextProps.hits), nextProps.aggregations);
 		}
+
+		if (
+			nextProps.defaultQuery
+			&& !isEqual(nextProps.defaultQuery(), this.defaultQuery)
+		) {
+			const options = getQueryOptions(nextProps);
+			options.from = this.state.from;
+			this.defaultQuery = nextProps.defaultQuery();
+
+			const { sort, ...query } = this.defaultQuery;
+
+			if (sort) {
+				options.sort = this.defaultQuery.sort;
+				nextProps.setQueryOptions(nextProps.componentId, options, !query);
```

Comment:
I don't think this will work as expected. `defaultQuery()` in case of ReactiveComponent should be executed on the internal component (for both `updateQuery()` and `setQueryOptions()`).

Reply:
Also, you should add support to execute query on any change in `queryOptions`, not just `sort`. It gets handled in a different manner in other components, but that's not the case here since this component is too vanilla as per its specs.